bigquery/google/cloud/bigquery/table.py

```diff
@@ -741,11 +741,9 @@ def insert_data(self,
             row_info = {}
 
             for field, value in zip(self._schema, row):
-                if field.field_type == 'TIMESTAMP':
-                    # BigQuery stores TIMESTAMP data internally as a
-                    # UNIX timestamp with microsecond precision.
-                    # Specifies the number of seconds since the epoch.
-                    value = _convert_timestamp(value)
+                converter = _SCALAR_VALUE_TO_JSON_ROW.get(field.field_type)
+                if converter is not None:
```

Comment:
I've added a comment.  As I said earlier, the `if` is waaaaaay cheaper than a no-op Python function call.